ext/drreg/drreg.c

```diff
@@ -410,7 +410,12 @@ drreg_insert_restore_all(void *drcontext, instrlist_t *bb, instr_t *inst,
     uint aflags = (uint)(ptr_uint_t)drvector_get_entry(&pt->aflags.live, pt->live_idx);
     if (!pt->aflags.native &&
         (force_restore ||
-         TESTANY(EFLAGS_READ_ARITH, instr_get_eflags(inst, DR_QUERY_DEFAULT)) ||
+         /* On AArch64, unrecognized instructions are decoded as instances of a generic
+          * instruction, OP_xx. For these instructions conservatively restore aflags in
+          * case they read/write the aflags.
+          */
```

Comment:
A workaround like this should be labeled as a temporary thing we do not want to leave in the code base long-term: so I would expect a XXX or TODO comment with an issue number covering the work to put in the permanent fix.

Reply:
I added a XXX comment linking to issue #2626 in case we decide to go ahead with this workaround till the decoder is more complete.